src/metadata.rs

```diff
@@ -32,12 +32,30 @@ impl TryFrom<Vec<u8>> for ProtocolVersion {
     }
 }
 
+impl fmt::Display for ProtocolVersion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let version = num::ToPrimitive::to_u64(self).ok_or(fmt::Error)?;
+        write!(f, "{}", version)
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
 
     #[test]
-    fn try_from_known_version() -> Result<(), ()> {
+    fn protocol_version_try_display() -> Result<(), ()> {
```

Comment:
```suggestion
    fn protocol_version_try_display() {
```

Reply:
This can get removed as we are not using `?` within the test blocks.

Reply:
Good catch, fixed and pushed